### PR TITLE
feat: input helper popover in dynamic form controls

### PR DESCRIPTION
**Pull request by dvh91**
This change is [Reviewable](https://reviewable.io/reviews/kaltura/kaltura-ng/36)

### CODE REVIEW

File: kaltura-ui/src/input-helper/input-helper.component.scss [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      &:hover, &.kActive {
+        color: $kSecondary;
+      }
+    }
+  }
+  
+  .kInputHelperPopup {
+    border-radius: 3px;
+    
+    /deep/ .kPopupWidget {
```
Comment (omridevk) [inline kaltura-ui/src/input-helper/input-helper.component.scss]: deep is deprecated, should use ng::deep according to angular doc:

> The shadow-piercing descendant combinator is deprecated and support is being removed from major browsers and tools. As such we plan to drop support in Angular (for all 3 of /deep/, >>> and ::ng-deep). Until then ::ng-deep should be preferred for a broader compatibility with the tools.
https://angular.io/guide/component-styles#deprecated-deep--and-ng-deep

**Comment by amirch1** [comment]
:lgtm:

---

Reviewed 6 of 9 files at r1, 1 of 2 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kaltura-ng/36#-:-L03vF7dRQHnFWVHcWWh:bnfp4nl)*

**Comment by amirch1** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[kaltura-ui/src/input-helper/input-helper.component.scss, line 24 at r1](https://reviewable.io:443/reviews/kaltura/kaltura-ng/36#-Kz3KK7H-RW2oSyX8rLz-r1-24:-L03vLKtMUNmR_IKE3Yv:b-1ze02c) ([raw file](https://github.com/kaltura/kaltura-ng/blob/17fd60fb8b13110196b590d5b57f888ac7fef138/kaltura-ui/src/input-helper/input-helper.component.scss#L24)):*
<details><summary><i>Previously, omridevk (Omri) wrote…</i></summary><blockquote>

deep is deprecated, should use ng::deep according to angular doc:

> The shadow-piercing descendant combinator is deprecated and support is being removed from major browsers and tools. As such we plan to drop support in Angular (for all 3 of /deep/, >>> and ::ng-deep). Until then ::ng-deep should be preferred for a broader compatibility with the tools.
https://angular.io/guide/component-styles#deprecated-deep--and-ng-deep


</blockquote></details>

Will be handled in our next libs upgrade across the entire app.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kaltura-ng/36)*

**Comment by esakal** [comment]
Reviewed 6 of 9 files at r1, 1 of 2 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[kaltura-ui/src/popup-widget/popup-widget.component.ts, line 58 at r3](https://reviewable.io:443/reviews/kaltura/kaltura-ng/36#-L042pk20dw2Oa58mNSJ:-L042pk20dw2Oa58mNSK:bq7e163) ([raw file](https://github.com/kaltura/kaltura-ng/blob/1347f766137ddefc32f108d828db3d52c0cd6b2a/kaltura-ui/src/popup-widget/popup-widget.component.ts#L58)):*
> ```XML
> 		if (this.trigger === 'click') {
> 			if (this._targetRef) {
> 				this._targetRef.removeEventListener('click', this.toggle.bind(this));
> ```

+@amirch1 please fix and push?

this is a memory leak because `bind` a function will always result with a new pointer.
```
const bla = [];
bla.forEach.bind(this) !== bla.forEach.bind(this)
```
see the following as a possible solution [link](https://stackoverflow.com/a/22870717/5874422)

---

*[kaltura-ui/src/popup-widget/popup-widget.component.ts, line 155 at r3](https://reviewable.io:443/reviews/kaltura/kaltura-ng/36#-L0441V98UH8HHf4W92B:-L0441V98UH8HHf4W92C:bp730ap) ([raw file](https://github.com/kaltura/kaltura-ng/blob/1347f766137ddefc32f108d828db3d52c0cd6b2a/kaltura-ui/src/popup-widget/popup-widget.component.ts#L155)):*
> ```XML
>             // handle modal
> 	        if (!this._modalOverlay) {
> 				if (this.trigger !== 'hover') {
> ```

this is breaking the ui. If the user ask to show 

---

*[kaltura-ui/src/popup-widget/popup-widget.component.ts, line 192 at r3](https://reviewable.io:443/reviews/kaltura/kaltura-ng/36#-L0450CfTYCmvmrfBDOd:-L0450CfTYCmvmrfBDOe:bb1jfp) ([raw file](https://github.com/kaltura/kaltura-ng/blob/1347f766137ddefc32f108d828db3d52c0cd6b2a/kaltura-ui/src/popup-widget/popup-widget.component.ts#L192)):*
> ```XML
> 		// auto positioning need first the dom to render
> 		setTimeout(() => {
> 			this.setPosition();
> ```

I think this is causing the following glitch in the ui? do you create it first below and then re-position it?

the provided animated gif only show one time of the glitch due to low capturing frequency, it happens every time showing that popup



---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kaltura-ng/36)*

**Comment by esakal** [comment]
Reviewed 2 of 3 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kaltura-ng/36)*

**Comment by esakal** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kaltura-ng/36#-:-L0o4lxB5m1-MCOQgi3j:bnfp4nl)*

**Pull request merged**